File: src/data/courses/dsa/two-pointers-sliding-window/two-pointers-opposite-direction/two-difference/code/python/two_difference_sol2.py
def find_pair(nums, target):
    nums.sort()
    i, j = 0, 1
    while i < len(nums) and j < len(nums):
        if i == j: j += 1; continue
        d = nums[j] - nums[i]
        if d == target: return True
        if d < target: j += 1
        else: i += 1
    return False

File: src/data/courses/dsa/two-pointers-sliding-window/two-pointers-opposite-direction/two-difference/code/cpp/two_difference_sol2.cpp
// Sort + Two Pointers: O(n log n)
bool findPair(vector<int>& nums, int target) {
    sort(nums.begin(), nums.end());
    int i = 0, j = 1;
    while (i < nums.size() && j < nums.size()) {
        if (i == j) { j++; continue; }
        int diff = nums[j] - nums[i];
        if (diff == target) return true;
        if (diff < target) j++;   // Difference too small
        else i++;                  // Difference too large
    }
    return false;
}